packages/plugin-legacy/src/index.ts

```diff
@@ -253,6 +257,22 @@ function viteLegacyPlugin(options: Options = {}): Plugin[] {
     },
   }
 
+  function sortPolyfills(
+    polyfillInfo: { chunkName: string; polyfills: string[] }[],
+  ): Set<string> {
+    const resultSet = new Set<string>()
+    polyfillInfo
+      .sort((a, b) =>
+        a.chunkName < b.chunkName ? -1 : a.chunkName > b.chunkName ? 1 : 0,
+      )
```

Comment:
We can't sort it this way because the chunk order matters and it would be good to respect that, hence I suggested using `meta.chunks` because it gives the right order or chunks.

This way we also prevent a large polyfill re-order change after this PR.

Reply:
Hi, the PR is updated as requested. Now the ordering of the chunks is taken from the forth parameter of `renderChunk`